packages/context/src/__tests__/unit/value-promise.unit.ts

```diff
@@ -31,6 +32,29 @@ describe('tryWithFinally', () => {
     expect(finalActionInvoked).to.be.true();
   });
 
+  it('skips error action for a fulfilled promise', async () => {
+    let errorActionInvoked = false;
+    const action = () => Promise.resolve(1);
+    const errorAction = (err: unknown) => {
+      errorActionInvoked = true;
+      throw err;
+    };
+    await tryCatchFinally(action, errorAction, () => {});
+    expect(errorActionInvoked).to.be.false();
+  });
+
+  it('skips error action for a resolved value', () => {
+    let errorActionInvoked = false;
+    const action = () => 1;
+    const errorAction = (err: unknown) => {
+      errorActionInvoked = true;
+      throw err;
+    };
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    tryCatchFinally(action, errorAction, () => {});
+    expect(errorActionInvoked).to.be.false();
+  });
+
```

Comment:
Please add tests to verify that `finally` callback is invoked both on success and on error.

I find it confusing that the tests for `tryCatchFinally` are in multiple places in the test file. Please reorganize them into their own `describe` block. Considering the already large size of this test file, it would be best to put the new tests into a new file, e.g. `packages/context/src/__tests__/unit/try-catch-finally.unit.ts`.